index.d.ts

```diff
@@ -0,0 +1,758 @@
+   * @param {Array} x an array (sorted)
+   * @param {*} v search value
+   * @param {function?} fn compare function (a, b)
+   * @returns {number} last index of value | ~(index of closest value)
+   */
+  export function bsearchRight<T>(x: T[], v: T, fn?: (a: T, b: T) => boolean): number;
+
+  /**
+   * Breaks array into chunks of given size.
+   * @param {Array} x an array
+   * @param {number?} n chunk size
+   * @returns {Array<Array>} chunks
+   */
+  export function chunk<T>(x: T[], n?: number): T[][];
+
+  /**
+   * Compares two arrays.
+   * @param {Array} x an array
+   * @param {Array} y another array
+   * @param {function?} fn compare function (a, b)
+   * @returns {number} x<y: -1, x=y: 0, x>y: 1
+   */
+  export function compare<T>(x: T[], y: T[], fn?: (a: T, b: T) => boolean): number;
+
+  /**
+   * Appends arrays to the end.
+   * @param {Array} x an array (updated)
+   * @param {...Iterable} ys arrays to append
+   * @returns {Array} x
+   */
+  export function concat$<T>(x: T[], ...ys: Iterable<T[]>[]): T[];
+
+  /**
+   * Gets true index to array (+ve).
+   * @param {Array} x an array
+   * @param {number} i index (+ve, -ve)
+   * @returns {number} +ve index
+   */
+  export function index<T>(x: T[], i: number): number;
+
+  /**
+   * Copies part of array to another.
+   * @param {Array} x target array (updated)
+   * @param {Array} y source array
+   * @param {number?} j write index (0)
+   * @param {number?} i read start index (0)
+   * @param {number?} I read end index (x.length)
+   * @returns {Array} x
+   */
+  export function copy$<T>(x: T[], y: T[], j?: number, i?: number, I?: number): T[];
+
+  /**
+   * Copies part of array to another.
+   * @param {Array} x target array
+   * @param {Array} y source array
+   * @param {number?} j write index (0)
+   * @param {number?} i read start index (0)
+   * @param {number?} I read end index (x.length)
+   * @returns {Array}
+   */
+  export function copy<T>(x: T[], y: T[], j?: number, i?: number, I?: number): T[];
+
+  /**
+   * Copies part of array within.
+   * @param {Array} x an array
+   * @param {number} j write index
+   * @param {number?} i read start index (0)
+   * @param {number?} I read end index (x.length)
+   * @returns {Array}
+   */
+  export function copyWithin<T>(x: T[], j: number, i?: number, I?: number): T[];
+
+  /**
+   * Counts occurrences of a value.
+   * @param {Array} x an array
+   * @param {*} v value
+   * @param {function?} fn compare function (a, b)
+   * @returns {number} occurrences
+   */
+  export function count<T>(x: T[], v: T, fn?: (a: T, b: T) => boolean): number;
+
+  /**
+   * Counts occurrences of values.
+   * @param {Array} x an array
+   * @param {function?} fn map function (v, i, x)
+   * @param {object?} ths this argument
+   * @returns {Map<any, number>} occurrences
+   */
+  export function countAllOn<T>(x: T[], fn?: (v: T, i: number, x: T[]) => T[], ths?: object): typeof Map;
```

Comment:
`map function` can return any new type, ex: map: string -> number.

```suggestion
  export function countAllOn<T,U>(x: T[], fn?: (v: T, i: number, x: T[]) => U, ths?: object): typeof Map<U,number>;
```

Reply:
Maybe `typeof` is not required.

Reply:
Removed `typeof`